### PR TITLE
feat(react-native-searchbox): add suggestion type query support

### PR DESCRIPTION
**Pull request by mohdashraf010897**
**PR Type** Feature Integration

**Description** Integration of new type of query `suggestion` to fetch suggestions of different types using a single API call.

- [x]  When we're making the suggestion query (whenever autosuggest is set to `true`, use the new suggestion type instead of the current manual calls for recent/popular suggestions.
- [x]  Remove manual API calls to `popular` suggestions and `recent` suggestions
- [x]  Update the popular/recent suggestions extraction logic from the suggestion API response
- [x] You don't need to display suggestions based on the value. The API would handle it. Currently, we display suggestions based on the value. For e.g, display recent suggestions when value is empty or popular and index suggestions when value is present.
- [x] Remove all the old prop names for supporting popular and recent suggestions.
- [x] Update docs.

[📄  Docs PR](https://github.com/appbaseio/Docs/pull/241) 

[📔  Notion ](https://www.notion.so/appbase/FE-Libraries-Integration-for-suggestion-type-cba01831fea4460d9639dd3df41043b0)

[Demo](https://snack.expo.dev/@mohdashraf010897/searchbox-react-native--alpha-realease-with-suggestions-integration)

### CODE REVIEW

File: packages/native/src/components/SearchBox/index.js [hunk 1]
```diff
@@ -358,8 +349,27 @@ class SearchBox extends React.Component {
...
+        '^(https?:\\/\\/)?' + // protocol
+        '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' + // domain name
+        '((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
+        '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
+        '(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
+          '(\\#[-a-z\\d_]*)?$',
+        'i'
+      ).test(suggestion.url)
+    ) {
+      Linking.openURL('https://google.com');
```
Comment (bietkul) [inline packages/native/src/components/SearchBox/index.js:365]: @mohdashraf010897 It should be the suggestion URL.

**Review by bietkul — CHANGES_REQUESTED** [review]

File: packages/native/src/components/SearchBox/index.js (hunk 1 above)
Comment (bietkul) [reply packages/native/src/components/SearchBox/index.js:365]:
```suggestion
      Linking.openURL(suggestion.url);
```

**Review by bietkul — APPROVED** [review]

File: packages/native/src/components/SearchBox/index.js (hunk 1 above)
Comment (bietkul) [inline packages/native/src/components/SearchBox/index.js]:
```suggestion
      Linking.openURL(suggestion.url);
```

**Review by bietkul — CHANGES_REQUESTED** [review]

File: packages/native/src/components/SearchBox/index.js (hunk 1 above)
Comment (mohdashraf010897) [reply packages/native/src/components/SearchBox/index.js]: @bietkul Resolved here ---> https://github.com/appbaseio/searchbox/pull/134/commits/85a35ea53234bbe88009b9b03401e96444caafbc

**Review by bietkul — APPROVED** [review]

**Pull request merged**